core/dbt/include/global_project/macros/materializations/helpers.sql

```diff
@@ -72,3 +72,141 @@
   {% endif %}
   {% do return(config_full_refresh) %}
 {% endmacro %}
+
+
+{% macro incremental_validate_on_schema_change(on_schema_change, default_value='ignore') %}
```

Comment:
There's enough here to warrant a new file. Let's move all of these macros (from here down) into `global_project/macros/materializations/incremental/on_schema_change.sql`

Reply:
Ok good deal, moving that code for postgres. I also got some good news during testing (on Snowflake, at least) that we essentially get the check on data type differences for free, because the `in` returns `False` when a checked column doesn't match the data type of what's in the reference list.